src/main/js/html.js

```diff
@@ -659,11 +660,16 @@
         return undefined;
     }
 
-    function spanMerge(first, second) {
+    function spanMerge(first, second, context) {
 
         if (first.tagName === "SPAN" &&
             second.tagName === "SPAN" &&
             first._isd_element === second._isd_element) {
+                if (! first._isd_element) {
+                    /* we should never get here since every span should have a source ISD element */
+                    reportError(context.errorHandler, "Span element is missing a source ISD element.");
+                    return false;
+                }
```

Comment:
Normally the error reporting is for errors in source content, but this looks like it is reporting an error in the code: is this really a change we want to commit, or a debugging artefact?

Reply:
> is this really a change we want to commit, or a debugging artefact?

I see three options:
- _throw_ since we should never get here (like an `assert`)
- report through the logger
- be silent

Let me know what you prefer.

Reply:
Out of those three, I think a log message is indeed probably appropriate here, as long as it is clear that the error is in the code rather than the source document. If I saw this when processing an IMSC document I would initially think that the document has some span element that is somehow incomplete: if I did not know better, and really looked hard, I might manage to navigate to https://www.w3.org/TR/ttml2/#isd-vocabulary-isd and then get _extremely_ confused!

Reply:
```suggestion
                if (! first._isd_element) {
                    /* we should never get here since every span should have a source ISD element */
                    reportError(context.errorHandler, "Internal error: HTML span is not linked to a source element; cannot merge spans.");
                    return false;
                }
```